src/config.rs

```diff
@@ -54,13 +54,32 @@ impl Config {
 
     pub(crate) fn list_toolchains(&self) -> Result<Vec<String>> {
         if self.toolchains_dir.is_dir() {
-            let mut toolchains: Vec<String> = fs::read_dir(&self.toolchains_dir)?
+            let mut custom_toolchains: Vec<String> = vec![];
+            let mut toolchains: Vec<String> = vec![];
+
+            for dir_entry in fs::read_dir(&self.toolchains_dir)?
                 .filter_map(io::Result::ok)
                 .filter(|e| e.file_type().map(|f| f.is_dir()).unwrap_or(false))
-                .map(|e| e.file_name().into_string().ok().unwrap_or_default())
-                .collect();
+            {
+                let toolchain = dir_entry.file_name().to_string_lossy().to_string();
+                if RESERVED_TOOLCHAIN_NAMES.iter().any(|t| {
+                    toolchain
+                        == format!(
+                            "{}-{}",
+                            t,
+                            &TargetTriple::from_host().unwrap_or_default().to_string()
```

Comment:
Do we have a function we can re-use for this toolchain-target-triple formatted string?

Reply:
Hmm, we only have 2 `format!`s here and both are used only within `config.rs`, so I didn't see the need for a function. I could create one within the same file.

Reply:
The idea of using a function is so that if the format changes in the future for whatever reason, we only have to change it in one location, and avoid running the risk of forgetting to update one of the locations.